src/Sarif/IFileRegionsCache.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public interface IFileRegionsCache
+    {
+        /// <summary>
+        /// Accepts a region, uri, and boolean and returns a Region object, based on the input
+        /// region property, that has all its properties populated. If an
+        /// input text region, for example, only specifies the startLine property, the returned
+        /// Region instance will have computed and populated other properties, such as charOffset,
+        /// charLength, etc.
+        /// </summary>
+        /// <param name="inputRegion">Input region to be based on.</param>
+        /// <param name="uri">Uri that will be used to get NewLineIndex from cache.</param>
+        /// <param name="populateSnippet">Boolean that indicates if the snipper will be populated.</param>
+        /// <returns>An instance of a Region class.</returns>
+        Region PopulateTextRegionProperties(Region inputRegion, Uri uri, bool populateSnippet);
+
+        /// <summary>
+        /// Construct new region based on input region and uri.
```

Comment:
>new region [](start = 22, length = 10)

What this method actually does is to create a _context region_ around the input region, and populate its code snippet. It probably should have been named `ConstructContextRegionWithSnippet`, but don't change that for now because it's a breaking change. Saying that it creates a "new region" doesn't tell the reader quite enough. So let's have the comment say "Create a context region around the specified region, including a code snippet" and the return value be "A context region around <param cref="inputRegion">."

Reply:
Other than that, it's fine.

---
In reply to: [511624864](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511624864) [](ancestors = 511624864)

Reply:
Oh darn it, wait. The interface itself needs a summary comment. In this case, the implementation class can't just `<inheritdoc/>` on the interface. I suggest that, just as the class summary comment starts with "A class that can be used to...", the interface summary comment can start with "An interface that can be used to..." and then the rest can be the same.

---
In reply to: [511624890](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511624890) [](ancestors = 511624890,511624864)

Reply:
Can you check again?

---
In reply to: [511625163](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511625163) [](ancestors = 511625163,511624890,511624864)

Reply:
Eddy, I'm sorry I wasn't clear about what comments to put where. Let me just pull your branch and tweak it.

---
In reply to: [511629640](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511629640) [](ancestors = 511629640,511625163,511624890,511624864)

Reply:
There you go.

---
In reply to: [511638795](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511638795) [](ancestors = 511638795,511629640,511625163,511624890,511624864)

Reply:
thank you! 

---
In reply to: [511641601](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511641601) [](ancestors = 511641601,511638795,511629640,511625163,511624890,511624864)